File: services/transaction-service/models/transferencia.go
package models

import (
	"github.com/google/uuid"
	"time"
)

type Estado string

const (
	Pendiente           Estado = "PENDIENTE"
	Procesando          Estado = "PROCESANDO"
	Completada          Estado = "COMPLETADA"
	Rechazada           Estado = "RECHAZADA"
	Compensando         Estado = "COMPENSANDO"
	Compensada          Estado = "COMPENSADA"
	CompensacionFallida Estado = "COMPENSACION_FALLIDA"
)

type Transferencia struct {
	IDTransferencia    uuid.UUID `json:"idTransferencia"`
	IDCliente          uuid.UUID `json:"idCliente"`
	IDCuentaOrigen     uuid.UUID `json:"idCuentaOrigen"`
	IDCuentaDestino    uuid.UUID `json:"idCuentaDestino"`
	IDCorrelacion      uuid.UUID `json:"idCorrelacion"`
	MontoCentavos      int64     `json:"montoCentavos"`
	Moneda             string    `json:"moneda"`
	Descripcion        string    `json:"descripcion,omitempty"`
	Estado             Estado    `json:"estado"`
	CodigoError        string    `json:"codigoError,omitempty"`
	FechaCreacion      time.Time `json:"fechaCreacion"`
	FechaActualizacion time.Time `json:"fechaActualizacion"`
}

File: frontend/bank-usac-web/src/features/transfers/types/transferencia.ts
export type EstadoTransferencia='VALIDANDO_KYC'|'VALIDANDO_CUENTAS'|'PENDIENTE'|'PROCESANDO'|'COMPLETADA'|'RECHAZADA'|'COMPENSANDO'|'COMPENSADA'|'COMPENSACION_FALLIDA';
export type ResultadoExternoSimulado='EXITO'|'FALLO'|'TIMEOUT';
export interface Transferencia{idTransferencia:string;idCliente:string;idCuentaOrigen:string;idCuentaDestino:string;idCorrelacion:string;montoCentavos:number;moneda:string;descripcion?:string;estado:EstadoTransferencia;codigoError?:string;resultadoExternoSimulado?:ResultadoExternoSimulado;fechaCreacion:string;fechaActualizacion:string}
export interface NuevaTransferencia{idCuentaOrigen:string;idCuentaDestino:string;tipoCuentaDestino:string;montoCentavos:number;descripcion?:string;resultadoExternoSimulado?:ResultadoExternoSimulado}
export interface OperacionAceptada{operationId:string;correlationId:string;status:string;statusUrl:string}
export interface EstadoOperacion{operationId:string;correlationId:string;type:string;status:string;updatedAt:string;errorCode?:string}
